jq.c

```diff
@@ -277,6 +289,96 @@ jdbc_detach_jvm()
 	(*jvm)->DetachCurrentThread(jvm);
 }
 
+static void jdbc_get_jni_env(void)
+{
+    int			JVMEnvStat;
+
+	ereport(DEBUG3, (errmsg("In In jdbc_get_jni_env")));
+
+    JVMEnvStat = (*jvm)->GetEnv(jvm, (void **) &Jenv, JNI_VERSION);
+    if (JVMEnvStat == JNI_EDETACHED)
+    {
+        ereport(DEBUG3, (errmsg("JVMEnvStat: JNI_EDETACHED; the current thread is not attached to the VM")));
+        jdbc_attach_jvm();
+    }
+    else if (JVMEnvStat == JNI_OK)
+    {
+        ereport(DEBUG3, (errmsg("JVMEnvStat: JNI_OK")));
+    }
+    else if (JVMEnvStat == JNI_EVERSION)
+    {
+        ereport(ERROR, (errmsg("JVMEnvStat: JNI_EVERSION; the specified version is not supported")));
+    }
+}
+
+static void jdbc_add_classpath_to_system_class_loader(char *classpath)
```

Comment:
You intentionally don't define and use the macros (e.g. `register_java_class`) defined in `scalardb_fdw`?

Reply:
Yes. This is because the original author doesn't use such a macro and calls the functions like `FindClass` directly. I intended to follow that way here.j